src/main/java/com/searchcode/app/jobs/DeleteRepositoryJob.java

```diff
@@ -32,25 +32,30 @@
 @DisallowConcurrentExecution
 public class DeleteRepositoryJob implements Job {
     public void execute(JobExecutionContext context) throws JobExecutionException {
+
         if (Singleton.getBackgroundJobsEnabled() == false) {
             return;
         }
 
         UniqueRepoQueue deleteRepoQueue = Singleton.getUniqueDeleteRepoQueue();
-        RepoResult rr = null;
+        RepoResult rr = deleteRepoQueue.poll();
+        if (rr == null) {
+            return;
+        }
+        // TODO Ruzzz Needed?
+        try {
+            Singleton.getUniqueGitRepoQueue().delete(rr);
+        } catch (Exception ex) {}
+        try {
+            Singleton.getUniqueSvnRepoQueue().delete(rr);
+        } catch (Exception ex) {}
+        try {
+            Singleton.getUniqueFileRepoQueue().delete(rr);
+        } catch (Exception ex) {}
```

Comment:
Can we be more explicit with the exception here. What is the exact exception we want to catch?